### PR TITLE
Makes `CliprdrBackend` `AsAny`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Teleport could use this feature, afaik it's harmless and simple to add (we're using `impl_as_any!`).

Corresponds with https://github.com/gravitational/teleport/pull/33506

### CODE REVIEW

File: crates/ironrdp-cliprdr/src/lib.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ enum CliprdrState {
 /// CLIPRDR static virtual channel client endpoint implementation
 #[derive(Debug)]
 pub struct Cliprdr {
-    backend: Box<dyn CliprdrBackend>,
+    pub backend: Box<dyn CliprdrBackend>,
```
Comment (pacmancoder) [inline crates/ironrdp-cliprdr/src/lib.rs:50]: Let's make getter for this instead

Comment (CBenoit) [reply crates/ironrdp-cliprdr/src/lib.rs:50]: What about exposing it with methods like these?

```rust
fn downcast_backend<T: Any>(&self) -> Option<&T> {
    self.backend.as_any().downcast::<T>()
}

fn downcast_backend_mut<T: Any>(&mut self) -> Option<&mut T> {
    self.backend.as_any_mut().downcast_mut::<T>()
}
```

Comment (ibeckermayer) [reply crates/ironrdp-cliprdr/src/lib.rs:50]: @pacmancoder sounds good.

@CBenoit I tried that initially but it breaks object safety (due to generics in trait methods) which is required for having `dyn CliprdrBackend`

Comment (ibeckermayer) [reply crates/ironrdp-cliprdr/src/lib.rs:50]: @CBenoit ah I see, adding those to `Cliprdr` itself 👍.

**Review by CBenoit — APPROVED** [review]
LGTM!

**Comment by CBenoit** [comment]
You need to implement `AsAny` for `WinCliprdrBackend` as well

**Pull request merged**